crates/nu-system/src/foreground.rs

```diff
@@ -98,15 +101,28 @@ impl Drop for ForegroundChild {
 #[cfg(unix)]
 mod foreground_pgroup {
     use nix::{
-        libc,
         sys::signal::{sigaction, SaFlags, SigAction, SigHandler, SigSet, Signal},
         unistd::{self, Pid},
     };
     use std::{
-        os::unix::prelude::CommandExt,
+        os::{
+            fd::{AsFd, BorrowedFd},
+            unix::prelude::CommandExt,
+        },
         process::{Child, Command},
     };
 
+    /// Alternative to having to call `std::io::stdin()` just to get the file descriptor of stdin
+    ///
+    /// # Safety
+    /// I/O safety of reading from `STDIN_FILENO` unclear.
+    ///
+    /// Currently only intended to access `tcsetpgrp` and `tcgetpgrp` with the I/O safe `nix`
+    /// interface.
+    pub unsafe fn stdin_fd() -> impl AsFd {
+        unsafe { BorrowedFd::borrow_raw(libc::STDIN_FILENO) }
```

Comment:
Oh cool, `tcgetpgrp` takes `impl AsFd` now. In that case, we can just use the standard lib `Stdin` type.

```suggestion
        std::io::stdin()
```

Reply:
The underlying [`impl AsFd for Stdin`](https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#423-428) does the same, but unsure if having to create [`Stdin` via `stdin`](https://doc.rust-lang.org/nightly/src/std/io/stdio.rs.html#332-339) can be optimized out?

Reply:
Oh ok, yeah it looks like it can't be [optimized out](https://godbolt.org/z/oxrrEo9hq).

Reply:
Thanks for godbolting! Funny how it completely extracts the constant into its `xor eax, eax` but is forced to keep the `call`